### ISSUE TITLE
Training tip for complex datasets

### ISSUE THREAD

**Issue by moiseshorta** [acids-ircam/RAVE#49]
Hi,

Just wanted to share some insights into training RAVE with complex datasets (full songs, heterogeneous sounds).

One solution I've come across having a good convergence on models is extending the `latent size` parameter to a full 128.
Also, which is what I think had a greater impact, is to extend the Phase 1 training or `warmup` to about 5 million steps before switching to Phase 2.

Obviously, more data will have better results, but so far I've found that the more 'detailed' features in my dataset, namely melodies, textures of individual mid-to-high freq. range instruments seem to start converging better beyond the suggested warmup phase of 1 million steps, as the loss keeps consistently going down.

Hope this helps anyone having trouble with this.

Good luck!

**Comment by vigliensoni** [comment]
Thank you for sharing, @moiseshorta!

Any insights on the number of steps for the adversarial fine-tuning and the latent representation compactness stages? 

Beyond `latent size`, have you seen any improvements changing other metaparameters?

**Comment by moiseshorta** [comment]
@vigliensoni so far I'm currently running a training experiment with the changed parameter, basically the modifications from the default parameters are:

```
--max-kl = 0.1 # effectively deactivates the beta scheduler
--warmup = 5000000
--latent-size = 128
```

Hope this helps!

**Comment by vigliensoni** [comment]
@moiseshorta, thx! also maintaining the `MAX_STEPS` last change of the `train_rave.py` file?

https://github.com/acids-ircam/RAVE/blob/b9e486570bfad18144f5b975734e9618967e1290/train_rave.py#L51-L51

**Comment by moiseshorta** [comment]
> @moiseshorta, thx! are you also maintaining the last change of `MAX_STEPS` in the `train_rave.py` file to try a longer fine tuning?
> 
> https://github.com/acids-ircam/RAVE/blob/b9e486570bfad18144f5b975734e9618967e1290/train_rave.py#L51-L51

Yes, I extended roughly double of the phase 1. So if phase 1 is 5M steps, the final count will be 10M

**Comment by wwerk** [comment]
Hi,
First of all, massive thanks for the tips - they will certainly save me from weeks of walking in the dark.

Considering that ~3h of material should probably do in case of more homogenous data, I was wondering what would be a reasonable size for a more complex dataset?

If more than 3h is necessary we are probably talking about sets containing whole subgenres :)

**Comment by caillonantoine** [comment]
So ignore my previous comment, `cli_helper` now ask you for a specific configuration ! You can also manually set RAVE's configuration using `python train_rave.py -c small,default,large` !
